tsconfig.json

```diff
@@ -8,8 +8,8 @@
     "module": "commonjs",
     // "esModuleInterop": true,
 
-    // Overrides default in order to remove "dom" because this package shouldn't assume the presence of browser APIs
-    "lib": ["ES2019"],
+    // Include `DOM` for access to the `fetch` global type definition. It won't be included in the built declaration files, but they'll assume the presence of that definition or an equivalent (ex. `@types/node`, `@cloudflare/workers-types`; see #401).
```

Comment:
Could we just declare this ourselves (`declare let fetch: (input: RequestInfo, init?: RequestInit) => Promise<Response>;`) for the assert on fetch-types.ts:74, or is this still necessary?

Reply:
RE both questions; the reason to do it this way is to keep the assertions up to date if the underlying libraries change. If we wanted to avoid this I’d just remove the relevant assertions entirely; there’s not much point in asserting that `1 === 1`.

If I were writing this library I’d drop the other 2 assertions (keeping the `node-fetch` ones because they don’t have downsides) and just be mindful if the fetch spec ever updates (rare and unlikely to affect the subset we use), but given that we already had assertions in the codebase I wasn’t sure if the repo’s stance was to keep those tests in place. LMK your take and I’ll update the PR :)

Reply:
I agree with the way you'd write it and low likelihood that the a fetch spec change would break this (or that @types/node will give us a better way to represent this by the time it does).

One question though - if we go this route, we can't rely on the `node-fetch` types to be compatible with DOM `fetch` right? AFAICT it's easy to update `headers` via `NonNullable<Record<string, string>>` but `body` is harder since `ReadableStream | XMLHttpRequestBodyInit` has no equivalent node types. I might be missing something there though

Reply:
I’ve removed the two extra assertions and added some clarifying comments ^_^

You are correct—`node-fetch`, DOM `fetch`, native Node fetch (via `undici`), Cloudflare fetch, etc. are all incompatible on the margins. Here’s [an insightful comment](https://github.com/microsoft/TypeScript/issues/43972#issuecomment-1202883038) that explains some of it. We just have to hope that future contributors only the common subset of fetch features across implementations (which, as above, is quite large and unlikely to regularly change).

Reply:
Got it, I see how you solved `body` (which was my primary concern) and left one comment on that part. In agreement on the rest of it!